### PR TITLE
update docker-compose file to use version 2 format

### PR DESCRIPTION
**Pull request by zubair-arbi**
ECOM-3180
@awais786 @ahsan-ul-haq @tasawernawaz @waheedahmed 

*  Update `docker-compose.yml` to use [version 2 format](https://docs.docker.com/compose/compose-file/#version-2)
* Add base credentials for mysql container

### CODE REVIEW

**Comment by waheedahmed** [comment]
lgtm :+1:

**Comment by zubair-arbi** [comment]
@clintonb please review

File: docker-compose.yml [hunk 1]
```diff
@@ -1,25 +1,36 @@
...
+  memcache:
+    image: memcached:1.4.24
+    container_name: memcache
 
-  # Uncomment the next two lines to build from a local configuration repo
-  #build: ../configuration
-  #dockerfile: docker/build/credentials/Dockerfile
+  credentials:
+    # Uncomment this line to use the official credentials base image
+    image: credentials:latest
```
Comment (clintonb) [inline docker-compose.yml]: This should be published to `edxops/credentials:latest`

**Comment by zubair-arbi** [comment]
@clintonb please review. I will upload the docker image to edxops docker hub after I have access. https://openedx.atlassian.net/browse/DEVOPS-4001

File: requirements/base.txt [hunk 2]
```diff
@@ -16,5 +16,6 @@ edx-rest-api-client==1.4.0
 libsass==0.10.0
 Pillow==3.0.0
 python-memcached==1.57
+python-social-auth==0.2.14
```
Comment (clintonb) [inline requirements/base.txt]: Add the note explaining that this should be removed when the PR merges.

**Comment by clintonb** [comment]
:+1:

**Pull request merged**